var/spack/repos/builtin/packages/py-nodeenv/package.py

```diff
@@ -12,8 +12,10 @@ class PyNodeenv(PythonPackage):
     homepage = "https://github.com/ekalinin/nodeenv"
     pypi = "nodeenv/nodeenv-1.3.3.tar.gz"
 
+    version("1.8.0", sha256="d51e0c37e64fbf47d017feac3145cdbb58836d7eee8c6f6d3b6880c5456227d2")
     version("1.7.0", sha256="e0e7f7dfb85fc5394c6fe1e8fa98131a2473e04311a45afb6508f7cf1836fa2b")
     version("1.3.3", sha256="ad8259494cf1c9034539f6cced78a1da4840a4b157e23640bc4a0c0546b0cb7a")
 
-    depends_on("py-setuptools", type="build")
+    depends_on("python@3.7:", when="@1.7:", type=("build", "run"))
```

Comment:
Python 2 is still supported. Honestly I would just remove this line

Reply:
I did not add python@2 because it is not supported in spack anymore. And after we/I broke the pipeline after removing a python@3.7 dependency I am also a little cautious about removing these dependencies

Reply:
The 3.7 deps are only required for style tools and their deps, which should be tested in CI. So unless CI fails, you can remove them without fear.

Reply:
Is removed now